providers/flagd/pkg/configuration.go

```diff
@@ -0,0 +1,116 @@
+	defaultPort                                              = 8013
+	defaultMaxEventStreamRetries                             = 5
+	defaultTLS                                        bool   = false
+	defaultCache                                      string = "lru"
+	defaultHost                                              = "localhost"
+	flagdHostEnvironmentVariableName                         = "FLAGD_HOST"
+	flagdPortEnvironmentVariableName                         = "FLAGD_PORT"
+	flagdTLSEnvironmentVariableName                          = "FLAGD_TLS"
+	flagdSocketPathEnvironmentVariableName                   = "FLAGD_SOCKET_PATH"
+	flagdServerCertPathEnvironmentVariableName               = "FLAGD_SERVER_CERT_PATH"
+	flagdCacheEnvironmentVariableName                        = "FLAGD_CACHE"
+	flagdMaxCacheSizeEnvironmentVariableName                 = "FLAGD_MAX_CACHE_SIZE"
+	flagdMaxEventStreamRetriesEnvironmentVariableName        = "FLAGD_MAX_EVENT_STREAM_RETRIES"
+	cacheDisabledValue                                       = cache.DisabledValue
+	cacheLRUValue                                            = cache.LRUValue
+	cacheInMemValue                                          = cache.InMemValue
+)
+
+type providerConfiguration struct {
+	CacheType                        string
+	CertificatePath                  string
+	EventStreamConnectionMaxAttempts int
+	Host                             string
+	MaxCacheSize                     int
+	OtelIntercept                    bool
+	Port                             uint16
+	SocketPath                       string
+	TLSEnabled                       bool
+}
+
+func newDefaultConfiguration() *providerConfiguration {
+	return &providerConfiguration{
+		CacheType:                        defaultCache,
+		EventStreamConnectionMaxAttempts: defaultMaxEventStreamRetries,
+		Host:                             defaultHost,
+		MaxCacheSize:                     defaultMaxCacheSize,
+		Port:                             defaultPort,
+		TLSEnabled:                       defaultTLS,
+	}
+}
+
+// updateFromEnvVar is a utility to update configurations based on current environment variables
+func (cfg *providerConfiguration) updateFromEnvVar(logger logr.Logger) {
+	portS := os.Getenv(flagdPortEnvironmentVariableName)
+	if portS != "" {
+		port, err := strconv.Atoi(portS)
+		if err != nil {
+			logger.Error(err,
+				fmt.Sprintf(
+					"invalid env config for %s provided, using default value: %d",
+					flagdPortEnvironmentVariableName, defaultPort,
+				))
+		} else {
+			cfg.Port = uint16(port)
+		}
+	}
+
+	host := os.Getenv(flagdHostEnvironmentVariableName)
+	if host != "" {
+		cfg.Host = host
+	}
+
+	socketPath := os.Getenv(flagdSocketPathEnvironmentVariableName)
+	if socketPath != "" {
+		cfg.SocketPath = socketPath
+	}
+
+	certificatePath := os.Getenv(flagdServerCertPathEnvironmentVariableName)
+	if certificatePath != "" || os.Getenv(flagdTLSEnvironmentVariableName) == "true" {
```

Comment:
maybe worth trimming and lowercase the value of `flagdTLSEnvironmentVariableName`

Reply:
not sure this is needed 🤔  I didn't want to add/change the behavior of this logic.

Reply:
Agree, this doesn't need to be part of this PR. However, maybe it is worth unifying across languages how to process bool env var. @toddbaert do we have somewhere in the spec a common way for SDKs to handle truthy and falsy values?